crates/pessimistic-proof/src/imported_bridge_exit.rs

```diff
@@ -36,24 +35,38 @@ impl ImportedBridgeExit {
         bridge_exit: BridgeExit,
         sending_network: NetworkId,
         imported_local_exit_root: KeccakDigest,
-        leaf_index: u32,
+        global_index: GlobalIndex,
         inclusion_proof: LETMerkleProof<Keccak256Hasher>,
+        inclusion_proof_rer: Option<(LETMerkleProof<Keccak256Hasher>, KeccakDigest)>,
     ) -> Self {
         Self {
             bridge_exit,
             sending_network,
-            imported_local_exit_root,
-            leaf_index,
+            global_index,
             inclusion_proof,
+            inclusion_proof_rer,
+            imported_local_exit_root,
         }
     }
 
     /// Verifies that the provided inclusion path is valid and consistent with the provided LER
     pub fn verify_path(&self) -> bool {
-        self.inclusion_proof.verify(
+        let leaf_inclusion = self.inclusion_proof.verify(
             self.bridge_exit.hash(),
-            self.leaf_index,
+            self.global_index.leaf_index,
             self.imported_local_exit_root,
+        );
+
+        self.inclusion_proof_rer.as_ref().map_or(
+            leaf_inclusion,
+            |(rollup_inclusion_proof, rollup_exit_root)| {
+                leaf_inclusion
+                    && rollup_inclusion_proof.verify(
+                        self.imported_local_exit_root,
+                        self.global_index.rollup_index,
+                        *rollup_exit_root,
+                    )
+            },
```

Comment:
In situations like this, I wish you could apply the `?` operator to a `bool` for early exit.

Maybe  this could be changed to return a `Result` with an error indicating which check failed and then use a `?`. Would it make sense or is it too much hassle for little benefit?

Reply:
Yeah better indeed, this function returns directly a `Result` now.